frontend/components/TableContainer/DataTable/DataTable.tsx

```diff
@@ -562,29 +563,53 @@ const DataTable = ({
                       (onSelectRowClick &&
                         disableMultiRowSelect &&
                         onSelectRowClick(row)) ||
-                        (onClickRow && onClickRow(row));
+                        (disableMultiRowSelect &&
+                          onClickRow &&
+                          onClickRow(row));
                     },
                     // For accessibility when tabable
                     onKeyDown: (e: KeyboardEvent) => {
                       if (e.key === "Enter") {
                         (onSelectRowClick &&
                           disableMultiRowSelect &&
                           onSelectRowClick(row)) ||
-                          (onClickRow && onClickRow(row));
+                          (disableMultiRowSelect &&
+                            onClickRow &&
+                            onClickRow(row));
                       }
                     },
                   })}
                   // Can tab onto an entire row if a child element does not have the same onClick functionality as clicking the whole row
                   tabIndex={keyboardSelectableRows ? 0 : -1}
                 >
-                  {row.cells.map((cell: any) => {
+                  {row.cells.map((cell: any, index: number) => {
+                    // Only allow row click behavior on first cell
```

Comment:
Will try to refactor later this week with another simplier solution (adding a stop propagation on the cells with checkboxes onclick. It'll get rid of this check on every cell. For now this will have to do.